driver/web/rest/adminapis.go

```diff
@@ -1387,3 +1387,460 @@ func (h AdminApisHandler) GetContentItemsCategories(claims *tokenauth.Claims, w
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	jsonData, err := json.Marshal(resData)
+	if err != nil {
+		log.Println("Error on marshal the updated content item")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonData)
+}
+
+// DeleteDataContentItem Deletes a data content item with a specified key
+// @Description Deletes a data content item with the specified key
+// @Tags Admin
+// @ID AdminDeleteDataContentItem
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/data/{id} [delete]
+func (h AdminApisHandler) DeleteDataContentItem(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+
+	err := h.app.Services.DeleteDataContentItem(claims, key)
+	if err != nil {
+		log.Printf("Error on deleting data content item with id - %s\n %s", key, err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+}
+
+// CreateCategory Creates a category
+// @Description Creates a category
+// @Tags Admin
+// @ID AdminCreateCategory
+// @Accept json
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/category [post]
+func (h AdminApisHandler) CreateCategory(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error on marshal create a category - %s\n", err.Error())
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	var item model.Category
+	err = json.Unmarshal(data, &item)
+	if err != nil {
+		log.Printf("Error on unmarshal the create a category - %s\n", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	createdItem, err := h.app.Services.CreateCategory(claims, &item)
+	if err != nil {
+		log.Printf("Error on creating category %s\n", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	jsonData, err := json.Marshal(createdItem)
+	if err != nil {
+		log.Println("Error on marshal the new category")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonData)
+}
+
+// GetCategory Gets a category
+// @Description Gets a category
+// @Tags Admin
+// @ID AdminGetCategory
+// @Accept json
+// @Produce json
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/category/{id} [get]
+func (h AdminApisHandler) GetCategory(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
```

Comment:
Let's make this path parameter `name` because that is what we are searching by in Mongo.